src/components/AccountPicker.jsx

```diff
@@ -1,69 +1,67 @@
+import { withClient } from 'cozy-client/dist/hoc'
+import { Overlay, Spinner } from 'cozy-ui/transpiled/react/'
 
-export const AccountPicker = ({
-  t,
-  connections,
-  history,
-  konnector,
-  match
-}) => {
-  const { konnectorSlug } = match.params
-  if (!connections.length)
-    return <Redirect to={`/connected/${konnector.slug}/new`} />
-  return (
-    <Modal dismissAction={() => history.push('/connected')} mobileFullscreen>
-      <ModalHeader>
-        <div className="col-account-connection-header">
-          <NavLink to="/connected" className="col-account-connection-back">
-            <Icon icon={backIcon} />
-          </NavLink>
-          <KonnectorHeaderIcon konnector={konnector} />
-        </div>
-      </ModalHeader>
-      <ModalContent>
-        <ul className={styles[classNames('col-account-picker')]}>
-          {connections.map((connection, index) => (
-            <li key={index}>
-              <AccountPickerItem
-                konnectorSlug={konnectorSlug}
-                connection={connection}
-                hasUpdate={hasPendingUpdate(konnector)}
-              />
-            </li>
-          ))}
-          <li>
-            <NavLink
-              to={`/connected/${konnectorSlug}/new`}
-              className={classNames(
-                styles['col-account-picker-button'],
-                styles['col-account-picker-button-add']
-              )}
-            >
-              <span>
-                <Icon icon="plus" />
-                <span>{t('account_picker.add_account_button.label')}</span>
-              </span>
-            </NavLink>
-          </li>
-        </ul>
-      </ModalContent>
-    </Modal>
-  )
+class AccountPicker extends Component {
+  state = {
+    triggersWithStatus: []
+  }
+  async componentDidMount() {
+    const { client, konnector } = this.props
+    const triggers = await client.query(
+      client
+        .find('io.cozy.triggers')
+        .where({
+          message: {
+            konnector: konnector.slug
+          },
+          worker: 'konnector'
+        })
+        .indexFields(['message.konnector', 'worker'])
+        .sortBy([{ 'message.konnector': 'desc' }, { worker: 'desc' }])
+    )
+    const triggersWithStatus = await Promise.all(
+      triggers.data.map(async trigger => {
+        const triggerWithStatus = await client.query(
+          client.get('io.cozy.triggers', trigger._id)
+        )
+        return triggerWithStatus.data
+      })
+    )
+    this.setState({ triggersWithStatus: { data: triggersWithStatus } })
+  }
+  render() {
+    const { connections, konnector, history } = this.props
+    const { triggersWithStatus } = this.state
+    const konnectorWithtriggers = { ...konnector, triggers: triggersWithStatus }
+    if (!connections.length)
+      return <Redirect to={`/connected/${konnector.slug}/new`} />
+    if (triggersWithStatus.length === 0)
+      return (
+        <Overlay className="u-flex u-flex-items-center u-flex-justify-center">
```

Comment:
Starting to wonder if a modal with a spinner inside wouldn't look better than this…